### PR TITLE
Added support for StorageDrawers and Deep Storage API

### PR DESCRIPTION
**Pull request by the-vindex**
Added wrapper around IInventory for integration with different storage systems. Used this for Storage Drawers and JABBA (general Deep Storage API).

I had to include Deep Storage API in Botania, which is quite ugly :-( But it will add mass support for all inventories using this API, which is apparently plenty.

TODO:
I don't have correct formatter for Botania - formatting may be off.

### CODE REVIEW

File: src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java [hunk 1]
```diff
@@ -167,69 +165,71 @@ public static int getCountInNetwork(ItemStack stack, Map<IInventory, Integer> in
...
+	 * purposes of counting huge amounts.
 	 */
-	public static List<ItemStack> requestItem(Object matcher, int itemCount, ICorporeaSpark spark, boolean checkNBT, boolean doit) {
-		List<ItemStack> stacks = new ArrayList();
+	public static List<ItemStack> requestItem(Object matcher, int itemCount, ICorporeaSpark spark, boolean checkNBT,
+			boolean doit) {
+		List<ItemStack> stacks = new ArrayList<ItemStack>();
 		CorporeaRequestEvent event = new CorporeaRequestEvent(matcher, itemCount, spark, checkNBT, doit);
-		if(MinecraftForge.EVENT_BUS.post(event))
+		if (MinecraftForge.EVENT_BUS.post(event))
```
Comment (TheWhiteWolves) [inline src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java]: remove all the spaces between your if and for statements and the brackets

File: src/main/java/vazkii/botania/api/package-info.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@API(owner = "Botania", apiVersion = "76", provides = "BotaniaAPI")
+@API(owner = "Botania", apiVersion = "77", provides = "BotaniaAPI")
```
Comment (TheWhiteWolves) [inline src/main/java/vazkii/botania/api/package-info.java]: Don't need to bump the API version as a release has not been made since it was increased to 76

**Comment by TheWhiteWolves** [comment]
As a general comment, do you need all of those files in the StorageDrawers API?, only include the ones that are required, we don't need to ship all the extra stuff

File: src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java (hunk 1 above)
Comment (the-vindex) [inline src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java]: I'll wait for proper Eclipse formatter in the repo. After that I'll reformat all my code with it so it matches what is correct. If you have it, please commit it to /eclipse folder and/or ping me on IRC.

**Comment by the-vindex** [comment]
We don't have to include all StorageDrawers API, I could try to cherrypick files from that API. On the other hand API is not included in JAR file, so it doesn't pollute runtime environment. So should I try to cherrypick?

**Comment by TheWhiteWolves** [comment]
you should, it may not be shipped with the jar but its included in the repo and pulled down to us, the less there is the better, means less stuff likely to need updating and less for us to manage.

**Comment by the-vindex** [comment]
I tried and it went very well. So done.

**Comment by Vazkii** [comment]
Proxy any API requests. The Botania API should not require others.

File: src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java (hunk 1 above)
Comment (TheWhiteWolves) [inline src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java]: you should be able to use the cleanup profile that is in the eclipse folder

Comment (the-vindex) [inline src/main/java/vazkii/botania/api/corporea/CorporeaHelper.java]: I have that imported, but clean up profile won't format the code (like those whitespaces between if and (). This is defined in Code Style = formatter.

**Comment by the-vindex** [comment]
Could you give me some example of what you mean by proxying?

**Comment by Vazkii** [comment]
https://github.com/Vazkii/Botania/blob/master/src/main/java/vazkii/botania/api/internal/IInternalMethodHandler.java#L118

**Comment by the-vindex** [comment]
Ok, done, new PR opened. Hope this is what you intended.

**Pull request closed without merging**